ext/mbstring/libmbfl/mbfl/mbfilter.c

```diff
@@ -855,8 +856,13 @@ mbfl_strpos(
 		needle_u8 = needle;
 	}
 
-	if (needle_u8->len < 1) {
-		result = (size_t) -8;
+	/* Check if offset is out of bound */
+	haystack_length = mbfl_strlen(haystack_u8);
```

Comment:
As mentioned before, mbfl_strlen() is expensive and we should not call it unless necessary. It's okay for the empty needle case because it's an edge-case (trying to include it in the code paths below would be better, but it's not really worth it), but it shouldn't be computed for each call. And I don't think it's necessary either, as out of bounds offsets will be detected in the loops below.

Reply:
ACK, will move the mbfl_strlen call back as a specialized call when the needle is empty and merge it when I come back from holidays.